### PR TITLE
[FIX] Dataset query to get only the latest facet for each version

### PR DESCRIPTION
**Pull request by sophiely**
### Problem

Closes: https://github.com/MarquezProject/marquez/issues/2860

### Solution

Since the same facet type is replicated a lot of times, we can rank the facet partition by dataset version and facet name ands so as we can take only the most recent facet for each dataset uuid and type.
The UI seems to display only one facet per type (facet name) and dataset version anyway so we don't need to query as much facet (which are just duplicates anyway).

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've included a one-line summary of your change for the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) (_Depending on the change, this may not be necessary_).
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [ ] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

**Review by phixMe — APPROVED** [review]
Amazing, thanks, do we have a similar problem elsewhere? Or is this the only instance of this problem that you have seen.

File: api/src/main/java/marquez/db/DatasetDao.java [hunk 1]
```diff
@@ -121,30 +121,51 @@ default void setFields(Dataset ds) {
...
-          WHERE d.namespace_name = :namespaceName
-          ORDER BY d.name
-          LIMIT :limit OFFSET :offset
-          """)
+      WITH facets_t AS
+          (SELECT df.dataset_version_uuid,
+                  df.facet,
+                  df."name",
+                  df.created_at,
+                  rank() OVER (PARTITION BY df.dataset_version_uuid, "name"
```
Comment (wslulciuc) [inline api/src/main/java/marquez/db/DatasetDao.java:129]: ❤️

**Review by wslulciuc — APPROVED** [review]
LGTM 💯 🚀 🔥

**Comment by dkt-sophie-ly** [comment]
> Amazing, thanks, do we have a similar problem elsewhere? Or is this the only instance of this problem that you have seen.

For now i don't think so but ofc I will let you know if I notice similar issues :)

**Pull request merged**